src/ctia/stores/es/init.clj

```diff
@@ -124,6 +119,21 @@
           (system-exit-error))
       existing)))
 
+(defn update-index-state
+  [{{update-mappings? :update-mappings
+     update-settings? :update-settings}
+    :props
+    :as conn-state}]
+  (when update-mappings?
+    (update-mappings! conn-state)
+    ;; template update must be after update-mapping
+    ;; if it fails a System/exit is triggered because
+    ;; this means that the mapping in invalid and thus
+    ;; must not be propagated to the template that would accept it
```

Comment:
I do not know how to properly test this comment and order, since it relies on the fact that `update-mappings` triggers a `System/exit` when the mapping is invalid (and thus prevent starting with a bad state). That's why the order matters.
Any thought on how to make this more robust is welcome.